scripts/src/models/orders.ts

```diff
@@ -240,24 +240,22 @@ export const Order = {
 	 * @param      {number}  limit
 	 * @return     {Promise<T[]>}  filtered orders including p2p
 	 */
-	async getAll<T extends Order>(filters: GetOrderFilters & { userId?: string }, limit?: number): Promise<T[]> {
-		const allOrders = await this.genericGet(filters).getMany(); // can be replaced by cache
-
-		const ids: string[] = allOrders.map(order => order.id);
+	async getAll<T extends Order>(filters: GetOrderFilters & ({ userId: string } | { walletAddress: string }), limit?: number): Promise<T[]> {
+		const contexts = await OrderContext.find(filters.userId ? { userId: filters.userId } : { wallet: filters.walletAddress }); // can be replaced by cache
+		const ids = contexts.map(c => c.orderId);
 		if (!ids.length) {
 			return []; // empty array causes sql syntax error
 		}
-
 		// remove all context related fields to get the second context
 		delete filters.userId;
 		delete filters.walletAddress;
+
 		const userOrdersQuery = this.genericGet(filters)
 			.andWhere(`ordr.id IN (:ids)`, { ids });
 
 		if (limit) {
-			userOrdersQuery.limit(limit);
+			userOrdersQuery.take(limit);
```

Comment:
using `.take(...)` instead of `.limit(...)` cases typeORM to be careful with relations when implementing pagination. so no more p2p order with only one context (it implements this though by issuing 2 queries instead of 1)